client/my-sites/checkout/composite-checkout/wpcom/components/wp-order-review-line-items.js

```diff
@@ -57,6 +57,19 @@ function WPLineItem( {
 	const isGSuite = !! item.wpcom_meta?.extra?.google_apps_users?.length;
 	const isSavings = item.type === 'coupon';
 
+	let domainDiscountCallout = null;
+	if ( item.wpcom_meta?.is_bundled && item.amount.value === 0 ) {
+		// Bundled domain registrations
+		domainDiscountCallout = (
+			<DiscountCalloutUI>{ translate( 'First year free' ) }</DiscountCalloutUI>
+		);
+	} else if ( item.wpcom_meta?.product_slug === 'domain_map' && item.amount.value === 0 ) {
```

Comment:
These `else if` sort of situations bother me because it's far too easy for this to grow and for the logic to get confusing in the future. Here's my suggestion: this code is creating a variable called `domainDiscountCallout` which has some logic and may or may not render something based on its inputs. That sounds an awful lot like a function call. And a function that renders something in React is a component. So if you moved this all into a function called `DomainDiscountCallout` and then below replaced `{ domainDiscountCallout }` with `<DomainDiscountCallout item={item} />`, I think it would be a lot more semantic. You'd also be able to avoid `else` by using an early return.